### PR TITLE
Fix to use NAT for Azure

### PR DESCRIPTION
**Pull request by tei-k**
# Description
https://scalar-labs.atlassian.net/browse/DLT-8152

# Done
- Add NAT for all subnets except `public`.
- Delete envoy-only NAT.

# Confirm
```
[centos@envoy-1 ~]$ chronyc sources
210 Number of sources = 4
MS Name/IP address         Stratum Poll Reach LastRx Last sample
===============================================================================
^- sh03.paina.net                1   6   377    61    -11ms[  -11ms] +/-   16ms
^- mail.somenet.org              2   6   377    62  -1497us[-1497us] +/-  161ms
^- mail.verbavatz.org            1   6   377    59  +1105us[+1105us] +/-  111ms
^* ntp-b2.nict.go.jp             1   6   377    64   +357us[ +434us] +/- 1521us

[centos@envoy-1 ~]$ curl inet-ip.info
20.46.188.11

[centos@cassandra-1 ~]$ curl inet-ip.info
20.46.188.11

[centos@scalardl-blue-1 ~]$ curl inet-ip.info
20.46.188.11

[centos@bastion-1 ~]$ curl inet-ip.info
23.102.76.218

$ telnet envoy-tei-lb-xnnrjc0.japaneast.cloudapp.azure.com 50051                                                                                                         main ✱
Trying 20.194.147.73...
Connected to envoy-tei-lb-xnnrjc0.japaneast.cloudapp.azure.com.
Escape character is '^]'.
```

### CODE REVIEW

File: modules/azure/scalardl/envoy.tf [hunk 1]
```diff
@@ -115,7 +87,7 @@ resource "azurerm_lb" "envoy_lb" {
 
   frontend_ip_configuration {
     name                          = "EnvoyLBAddress"
-    public_ip_address_id          = local.envoy.nlb_internal ? "" : azurerm_public_ip.envoy_public_ip.*.id[count.index]
+    public_ip_address_id          = local.envoy.nlb_internal ? "" : azurerm_public_ip.envoy_public_ip[count.index].id
```
Comment (feeblefakie) [inline modules/azure/scalardl/envoy.tf:90]: Is this change related to NAT thing? or just an improvement?
If it is related, then it's fine.
if it is just an improvement, it would be great if it is separated since NAT thing is an enhancement.

**Review by feeblefakie — APPROVED** [review]
LGTM! Left one question.

File: modules/azure/network/main.tf [hunk 2]
```diff
@@ -33,6 +33,32 @@ resource "azurerm_subnet" "subnet" {
...
+
+resource "azurerm_nat_gateway" "nat_gateway" {
+  name                    = "nat-gateway"
+  location                = var.region
+  resource_group_name     = azurerm_resource_group.resource_group.name
+  public_ip_address_ids   = [azurerm_public_ip.nat_ip.id]
+  sku_name                = "Standard"
+  idle_timeout_in_minutes = 10
+}
+
```
Comment (tei-k) [inline modules/azure/network/main.tf:52]: 📝 It is best to create a nat for each az. I'll implement this when major refactoring.

File: modules/azure/scalardl/envoy.tf [hunk 3]
```diff
@@ -125,43 +97,56 @@ resource "azurerm_lb_backend_address_pool" "envoy_lb_pool" {
...
+
+resource "azurerm_lb_rule" "envoy_lb_udp_rule" {
+  count = local.envoy.enable_nlb ? 1 : 0
+
+  resource_group_name            = local.network_name
+  loadbalancer_id                = azurerm_lb.envoy_lb[count.index].id
+  name                           = "EnvoyLBUdpRule"
+  protocol                       = "Udp"
+  frontend_port                  = 123
+  backend_port                   = 123
```
Comment (tei-k) [inline modules/azure/scalardl/envoy.tf:140]: `123` is dummy port, just for allow udp outbound with chrony.

File: modules/azure/scalardl/envoy.tf (hunk 1 above)
Comment (tei-k) [reply modules/azure/scalardl/envoy.tf:90]: I added resource `envoy_lb_udp_rule` with new style. It's mixed up with the old style, so I fixed it while I was at it.

**Review by ymorimo — APPROVED** [review]
LGTM!
